### PR TITLE
chore(test): auto retest prow

### PR DESCRIPTION
**Pull request by janisz**
## Description

This PR creates an action that will run every hour and scan PRs with `auto-merge` label and if there is a failed status then comment with `/retest`. This is a first step into making an action that will run prow tests on PR with given number of times. Currently there is a hard limit of 10 retests.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### Here I tell how I validated my change

Run on [`pull_request` event](https://github.com/stackrox/stackrox/actions/runs/8359198489/job/22882040137#step:4:73)
that resulted with following comment
- https://github.com/stackrox/stackrox/pull/10404#issuecomment-2009492469
- https://github.com/stackrox/stackrox/actions/runs/8375351851/job/22932512422?pr=10438

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

File: tools/retest/main.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+			if c.GetBody() == "/retest" {
+				retested++
+			}
+		}
+		log.Printf("#%d was retested %d times", prNumber, retested)
+		if retested > 10 {
+			continue
+		}
+
+		get, _, err := client.PullRequests.Get(ctx, "stackrox", "stackrox", prNumber)
```
Comment (porridge) [inline tools/retest/main.go:40]:
```suggestion
		prDetails, _, err := client.PullRequests.Get(ctx, "stackrox", "stackrox", prNumber)
```

File: tools/retest/main.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+		handleError(err)
+
+		var statuses []Status
+		request, err := http.NewRequest("GET", get.GetStatusesURL(), nil)
```
Comment (porridge) [inline tools/retest/main.go:44]:
```suggestion
		statusRequest, err := http.NewRequest("GET", get.GetStatusesURL(), nil)
```

File: tools/retest/main.go [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+		handleError(err)
+		_, err = client.Do(ctx, request, &statuses)
+		handleError(err)
+
+		retest := "/retest"
```
Comment (porridge) [inline tools/retest/main.go:49]: Use this variable in the `==` check above too?

File: tools/retest/main.go [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+		retestComment := github.IssueComment{
+			Body: &retest,
+		}
+
+		for _, status := range statuses {
```
Comment (porridge) [inline tools/retest/main.go:54]: I don't get this loop. Can there be more than one status? What are these? Do we want to really issue `/retest` multiple times if there is more than one `failure` status?

**Review by porridge — APPROVED** [review]
I think this is good enough as a prototype.
Some comments inline, mostly to rename some variables.

File: tools/retest/main.go (hunk 4 above)
Comment (janisz) [reply tools/retest/main.go:55]: Ah, that's good point!

File: tools/retest/main.go (hunk 3 above)
Comment (janisz) [reply tools/retest/main.go]: ok, but it cannot be a const as we need a pointer here

**Pull request merged**